### PR TITLE
Move WPCS Deprecated VIP Sniffs to VIPCS.

### LINKED ISSUE #187: WPCS 2.0: Migrate Deprecated VIP Sniffs
WPCS 1.0.0 deprecates VIP sniffs, with the intention that they be removed in a later release. We should copy these sniffs to VIP Minimum appropriatley so that our own sniffs and rules aren't broken by a future release

### LINKED ISSUE #298: WPCS 2.0: Remove `WordPress.VIP` references from rulesets
Extracted from #187.

WPCS has just [removed the VIP sniffs](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1551) from it's `develop` branch, so we need to remove references to `WordPress.VIP` in our rulesets.

The `WordPressVIPMinimum` [`ruleset.xml`](https://github.com/Automattic/VIP-Coding-Standards/blob/dae432a40e20ccc4f2122c9b950a6ab510d4ec81/WordPressVIPMinimum/ruleset.xml) includes the following `WordPress.VIP` references:

```xml
	<rule ref="WordPress.VIP">
		<exclude name="WordPress.VIP.ValidatedSanitizedInput.MissingUnslash"/>
		<exclude name="WordPress.VIP.RestrictedFunctions"/>
	</rule>
...
	<rule ref="WordPress.VIP.AdminBarRemoval.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.CronInterval.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.DirectDatabaseQuery.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.FileSystemWritesDisallow.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.OrderByRand.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.PluginMenuSlug.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPressVIPMinimum.VIP.RestrictedFunctions.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.RestrictedVariables.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.SessionFunctionsUsage.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.SessionVariableUsage.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.SlowDBQuery.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.SuperGlobalInputUsage.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.TimezoneChange.DeprecatedSniff">
		<severity>0</severity>
	</rule>
	<rule ref="WordPress.VIP.ValidatedSanitizedInput.DeprecatedSniff">
		<severity>0</severity>
	</rule>
```

----

The `WordPress-VIP-Go` [`ruleset.xml`](https://github.com/Automattic/VIP-Coding-Standards/blob/dae432a40e20ccc4f2122c9b950a6ab510d4ec81/WordPress-VIP-Go/ruleset.xml) includes the following `WordPress.VIP` references:

```xml
	<rule ref="WordPress.VIP.RestrictedFunctions.get_posts">
		<severity>1</severity>
	</rule>	
...
	<rule ref="WordPress.VIP.SuperGlobalInputUsage.AccessDetected">
		<type>warning</type>
	</rule>
```

### PR DESCRIPTION
**Pull request by rebeccahum**
This PR:

* Adds the WPCS deprecated VIP Sniffs (except for `SuperGlobalInputUsage` because `ValidatedSanitizedInput` flags the same variables)  and their related unit tests to the VIPCS VIP Category:
    * `AdminBarRemovalSniff`
    * `CronIntervalSniff`
    * `FileSystemWritesDisallowSniff`
    * `OrderByRandSniff`
    * `NoPagingSniff` (formerly known as `PostsPerPage` but WPCS has a sniff now for it)
    * `RestrictedFunctionsSniff`
    * `RestrictedVariablesSniff`
    * `SessionFunctionsUsageSniff`
    * `SessionVariableUsageSniff`
* Adds to the `WordPressVIPMinimum` ruleset to reference the WPCS sniffs (as they are no longer categorized in WPCS VIP):
    * `WordPress.Security.ValidatedSanitizedInput`
    * `WordPress.Security.PluginMenuSlug`
    * `WordPress.WP.PostsPerPage`
    * `WordPress.WP.TimezoneChange`
    * `WordPress.DB.DirectDatabaseQuery`
    * `WordPress.DB.SlowDBQuery`
* Adjusts the `WordPress.VIP` references in the rulesets to use `WordPressVIPMinimum`.
* Removes the silencing of the deprecated WPCS deprecated VIP sniffs.
* Fixes the PHPDoc comments on existing VIPCS sniffs in the VIP Category.

In short, it fixes #298 and resolves #187 as well.

### CODE REVIEW

File: WordPressVIPMinimum/Sniffs/VIP/NoPagingSniff.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Flag returning high or infinite posts_per_page.
+ *
+ * @link https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/#no-limit-queries
+ *
+ * @package VIPCS\WordPressVIPMinimum
+ *
+ * @since   0.5.0
+ */
+class PostsPerPageSniff extends AbstractArrayAssignmentRestrictionsSniff {
```
Comment (GaryJones) [inline WordPressVIPMinimum/Sniffs/VIP/NoPagingSniff.php]: Class name here doesn't match the file name.

File: WordPressVIPMinimum/Tests/VIP/AdminBarRemovalUnitTest.inc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+add_filter( 'show_admin_bar', '__return_false' ); // Bad.
+add_filter( 'show_admin_bar', '__return_true' ); // Ok.
+
+show_admin_bar( false ); // Bad.
+show_admin_bar( true ); // Ok.
+
+add_filter( 'show_admin_bar', 'my_own_return_false' ); // Bad.
+
+// @codingStandardsChangeSetting WordPressVIPMinimum.VIP.AdminBarRemoval remove_only false
```
Comment (GaryJones) [inline WordPressVIPMinimum/Tests/VIP/AdminBarRemovalUnitTest.inc]: These annotations will need updating - see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1571

All of the .inc files in this PR will need checking and fixing up please.

**Comment by GaryJones** [comment]
@rebeccahum 

I rebased the PR to take fix the merge conflict due to the removal of FlushRewriteRulesSniff, and I also refreshed the integration test files as overlapping work from #309.

I've expanded on the commit messages as well, to give them more context.

**Comment by GaryJones** [comment]
Audit updated. 

See commit messages for Change Log info.
